python/cuml/common/input_utils.py

```diff
@@ -403,6 +403,28 @@ def input_to_cupy_array(X,
     Identical to input_to_cuml_array but it returns a cupy array instead of
     CumlArray
     """
+    if isinstance(X, cudf.DataFrame):
+        # Columns containing missing values
+        colWithMissing = X.isnull().any().to_array()
+        if colWithMissing.any():
+            # Columns types
+            colTypes = X.dtypes.values
+            # Types of columns containing missing values
+            colTypesWithMissing = colTypes[colWithMissing]
+            checkIfFloatType = np.vectorize(
+                lambda x: not np.issubclass_(x, float))
+            # Columns that require conversion
+            colToConvert = checkIfFloatType(colTypesWithMissing)
```

Comment:
isn't this detection of missing values way easier with `.values`?:

```python
>>> df = cudf.DataFrame(data=[[7, 2, 3], [4, None, 6], [10, 5, 9]])
>>> df
    0     1  2
0   7     2  3
1   4  <NA>  6
2  10     5  9
>>> df.values
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/galahad/miniconda3/envs/ns1116/lib/python3.8/site-packages/cudf/core/dataframe.py", line 926, in values
    return cupy.asarray(self.as_gpu_matrix())
  File "/home/galahad/miniconda3/envs/ns1116/lib/python3.8/site-packages/cudf/core/dataframe.py", line 3411, in as_gpu_matrix
    raise ValueError(
ValueError: column '1' has null values. hint: use .fillna() to replace null values
```

Reply:
Do you want me to catch the Exception?

Reply:
I am afraid that it might be slower in case the Dataframe is very large and is successfully returned to host.

Reply:
May be done with a `fillna` call however.

Reply:
If I'm not mistaken, the cudf code for `as_gpu_matrix` (which `.values` calls) does a quick check for each column to see if they have a null count (which if they have nulls it is already computed), so I think it shouldn't do any copying/converting before checking for nulls https://github.com/rapidsai/cudf/blob/c34d9bf04a71da3fed07f1f564a715f379ac7676/python/cudf/cudf/core/dataframe.py#L3410 (I would double check that though)

Reply:
Ah but `.values` only works for numeric columns, is that an issue?

Reply:
Ah, yes `.values` returns a cuPy array, so no host transfer. Yes, I think you are right it does the [check](https://github.com/rapidsai/cudf/blob/bd564a0102a5c8b6bd34ff1c2592f42d76fe55da/python/cudf/cudf/core/dataframe.py#L3409) before copying. I think the call to `input_to_cuml_array` is still necessary to perform checks.